propagation/env.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package propagation // import "go.opentelemetry.io/otel/propagation"
+
+import (
+	"os"
+	"strings"
+)
+
+// EnvCarrier is a TextMapCarrier that uses the environment variables as a
+// storage medium for propagated key-value pairs. The keys are uppercased
+// before being used to access the environment variables.
+// This is useful for propagating values that are set in the environment
+// and need to be accessed by different processes or services.
+// The keys are uppercased to avoid case sensitivity issues across different
+// operating systems and environments.
+type EnvCarrier struct{}
+
+var _ TextMapCarrier = EnvCarrier{}
+
+// Get returns the value associated with the passed key.
+// The key is uppercased before being used to access the environment variable.
+func (EnvCarrier) Get(key string) string {
+	k := strings.ToUpper(key)
+	return os.Getenv(k)
+}
+
+// Set stores the key-value pair in the environment variable.
+// The key is uppercased before being used to set the environment variable.
+func (EnvCarrier) Set(key, value string) {
+	k := strings.ToUpper(key)
+	_ = os.Setenv(k, value)
+}
+
+// Keys lists the keys stored in this carrier.
+// This method is not implemented for EnvCarrier as it is not possible to
+// list all environment variables in a portable way.
+func (EnvCarrier) Keys() []string {
+	// I don't know why TextMapCarrier even has a Keys method.
+	// It looks like it was some mistake in the original design.
+	return nil
+}
```

Comment:
@open-telemetry/go-maintainers, do you know why `TextMapCarrier` has a `Keys` method? It does not look useful at all. I think about adding a comment that we have it just for backwards compatibility...

I have not seen `Keys` used anywhere besides tests. 

I saw [this](https://github.com/open-telemetry/opentelemetry-specification/blob/9eee5293f95b9fd74f6f1c280b97f87aaec872d7/specification/context/api-propagators.md#L135-L136):

> If your carrier is reused, you should delete the fields here before calling Inject.

However, I do not see sense in it.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/6808